src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2934,7 +2934,8 @@ public IEnumerable<ISymbol> GetSymbolsWithName(Func<string, bool> predicate, Sym
                 var result = new HashSet<ISymbol>();
                 var spine = new List<MergedNamespaceOrTypeDeclaration>();
 
-                AppendSymbolsWithName(spine, _compilation.Declarations.MergedRoot, predicate, filter, result, cancellationToken);
```

Comment:
Should we rename MergedRoot to MergedRootUnordered to make it explicit that one should not assume any particular order?